lib/blemulator/internal/bridge/dart_to_platform_bridge.dart

```diff
@@ -1,29 +1,54 @@
 part of internal;
 
 class DartToPlatformBridge {
-  MethodChannel _downstreamChannel;
+  MethodChannel _dartToPlatformChannel;
 
   DartToPlatformBridge() {
-    _downstreamChannel = new MethodChannel(ChannelName.downstream);
+    _dartToPlatformChannel = new MethodChannel(ChannelName.dartToPlatform);
   }
 
   Future<void> simulate() =>
-      _downstreamChannel.invokeMethod(PlatformMethodName.simulate);
+      _dartToPlatformChannel.invokeMethod(PlatformMethodName.simulate);
 
-  Future<void> addScanResult(ScanResult scanResult) =>
-      _downstreamChannel.invokeMethod(
+  Future<void> publishScanResult(ScanResult scanResult) =>
+      _dartToPlatformChannel.invokeMethod(
           SimulationPlatformMethodName.publishScanResult, <String, dynamic>{
-        ArgumentName.name: scanResult.name,
-        ArgumentName.id: scanResult.id,
-        ArgumentName.mtu: scanResult.mtu,
-        ArgumentName.rssi: scanResult.rssi,
-        ArgumentName.isConnectable: scanResult.isConnectable,
-        ArgumentName.txPowerLevel: scanResult.txPowerLevel,
-        ArgumentName.manufacturerData: scanResult.manufacturerData,
-        ArgumentName.serviceData: scanResult.serviceData,
-        ArgumentName.serviceUuids: scanResult.serviceUuids,
-        ArgumentName.localName: scanResult.localName,
-        ArgumentName.solicitedServiceUuids: scanResult.solicitedServiceUuids,
-        ArgumentName.overflowUuids: scanResult.overflowUuids,
+        SimulationArgumentName.name: scanResult.name,
+        SimulationArgumentName.id: scanResult.id,
+        SimulationArgumentName.mtu: scanResult.mtu,
+        SimulationArgumentName.rssi: scanResult.rssi,
+        SimulationArgumentName.isConnectable: scanResult.isConnectable,
+        SimulationArgumentName.txPowerLevel: scanResult.txPowerLevel,
+        SimulationArgumentName.manufacturerData: scanResult.manufacturerData,
+        SimulationArgumentName.serviceData: scanResult.serviceData,
+        SimulationArgumentName.serviceUuids: scanResult.serviceUuids,
+        SimulationArgumentName.localName: scanResult.localName,
+        SimulationArgumentName.solicitedServiceUuids:
+            scanResult.solicitedServiceUuids,
+        SimulationArgumentName.overflowUuids: scanResult.overflowUuids,
       });
+
+  Future<void> publishConnectionState(SimulatedPeripheral peripheral,
+          PeripheralConnectionState connectionState) =>
+      _dartToPlatformChannel.invokeMethod(
+          SimulationPlatformMethodName.publishConnectionState,
+          <String, dynamic>{
+            SimulationArgumentName.id: peripheral.id,
+            SimulationArgumentName.connectionState: _connectionStateToString(connectionState),
+          });
+
+  String _connectionStateToString(PeripheralConnectionState state) {
+    switch (state) {
+      case PeripheralConnectionState.connecting:
+        return "CONNECTING";
```

Comment:
We already have NativeConnectionState class for these values.

Reply:
We shouldn't use internals of FlutterBleLib inside BLEmulator. I've created a separate definition for those values in this module.